rossum_api/elis_api_client_sync.py

```diff
@@ -75,29 +77,40 @@ def __init__(
         self.elis_api_client = ElisAPIClient(
             username, password, token, base_url, http_client, deserializer
         )
-
-        try:
-            self.event_loop = asyncio.get_running_loop()
-            if self.event_loop.is_running():
-                raise AsyncRuntimeError(
-                    "Event loop is present and already running, please use async version of the client"
-                )
-        except RuntimeError:
-            self.event_loop = asyncio.new_event_loop()
+        self.executor = ThreadPoolExecutor()
 
     def _iter_over_async(self, ait: AsyncIterator[T]) -> Iterator[T]:
-        ait = ait.__aiter__()
-        while True:
+        """Iterate over an async generator from sync code without materializing all items into memory."""
+        queue: ThreadSafeQueue = (
+            ThreadSafeQueue()
+        )  # To communicate with the thread executing the async generator
+
+        async def async_iter_to_list(ait: AsyncIterator[T], queue: ThreadSafeQueue):
             try:
-                obj = self.event_loop.run_until_complete(ait.__anext__())
-                yield obj
-            except StopAsyncIteration:
+                async for obj in ait:
+                    queue.put(obj)
+            finally:
+                queue.put(None)  # Signal iterator was consumed
+
+        future = self.executor.submit(asyncio.run, async_iter_to_list(ait, queue))  # type: ignore
+
+        # Consume the queue until completion to retain the iterator nature even in sync context
+        while True:
+            item = queue.get()
+            if item is None:  # None is used to signal completion
                 break
+            yield item
+
+        future.result()
```

Comment:
This is probably just an idea for future improvement, but should be rossum-api be able to set a total timeout?
We would have to replicate it in the sync client.

Reply:
There are timeouts for individual HTTP requests which is probably enough.